Support JSX fragments with jsxFragFactory compiler option and @jsxFrag pragma (#35392)
microsoft/TypeScript: tests/baselines/reference/jsxFactoryAndFragment.errors.txt

> tests/cases/compiler/jsxFactoryAndFragment.tsx(3,1): error TS17016: JSX fragment needs corresponding `jsxFragFactory` compiler option when using `jsxFactory` compiler option.

Reviewer: ```suggestion
tests/cases/compiler/jsxFactoryAndFragment.tsx(3,1): error TS17016: The `jsxFragmentFactory` compiler option must be provided to use JSX fragments with the `jsxFactory` compiler option.
```
